libsg3/ses/ses.pyx

```diff
@@ -329,10 +329,17 @@ cdef class EnclosureDevice(object):
         }
         return element_dict[index]
 
-    def decode_response(self, buff):
+    cdef str decode_response(self, char * buff):
+        cdef int i = 0;
+        with nogil:
+            while buff[i] != 0:
+                if buff[i] < 0 or buff[i] > 127:
+                    buff[i] = 26    # SUBstitute character
+                i += 1;
         try:
             return bytes(buff, encoding='ascii').decode()
-        except Exception:
+        except Exception as e:
+            raise(e)
             return ""
```

Comment:
Will we ever reach `return` after `raise`?

Reply:
I think it's not needed. I have removed it.